aoc_day_2/src/part2.rs

```diff
@@ -42,7 +42,6 @@ where P: AsRef<Path>, {
 
 fn calculatewinning_score(outcomes: &mut Vec<&str>) -> u32 {
     
-    
     let mut score: u32 = 0;
     let mut subject_move = outcomes[1];
     
```

Comment:
Some notes on the below: good job using matches in your refactoring! you still use _if blocks_ over an exhaustive list, so let me just parrot what I have absorbed (rightly or wrongly) to be the idiomatic rust way.

When your if statements consider conditions that compare an exhaustive subset .. I mean, well, you could make an enum out of something in the conditional, like:

```rust
enum Outcomes {
  Win,
  Lose.
  Draw
}

impl Outcomes {
  pub fn from_string(s: String) {
    match s {
      "X" => Outcomes::Lose,
      "Y" => Outcomes::Draw,
      "Z" => Outcomes::Win,
      _ => unimplemented!()
    }
  }
}
```

so your if statements consider "X", "Y", and "Z", as they are bound to the Outcomes domain. instead of an if statement, you could use a _match_ to ensure that as your domain changes, your code immediately tells you when you haven't considered all the consequences:

```rust
    match Outcomes::from_string(s) {
      Outcomes::Lose => {
        subject_move = match outcomes[0] {
            "A" => "S",
            "B" => "R",
            "C" => "P",
            _ => subject_move,
        };
      },
      Outcomes::Draw => { ... },
      Outcomes::Win => { ... }
    }
```

you don't have to create the enums but it can help. Like in your code here:

```rust
        subject_move = match outcomes[0] {
            "A" => "S",
            "B" => "R",
            "C" => "P",
            _ => subject_move,
        };
```

the domain you are considering is indexed by "A" | "B" | "C" (like, in typescript, that could be your type). So the catch all case you needed to add could be obviated by using an enum instead of raw strings. OR, you can just approximate it with 

```rust
_ => unimplemented!()
```

The bigger issue though is that there is code repetition in:
```rust
    match (outcomes[0], subject_move) {
        ("A", "R") => score += 3,
        ("A", "P") => score += 6,
        ("A", "S") => score += 0,
        ("B", "S") => score += 6,
        ("B", "P") => score += 3,
        ("B", "R") => score += 0,
        ("C", "R") => score += 6,
        ("C", "P") => score += 0,
        ("C", "S") => score += 3,
        _ => (),
    }
```

as far as it goes, not really such a big deal honestly :)

if multiple keys point to the same value, they must be equivalent. this match should only have 3 branches (with a 4th for `_ => unimplemented!()` if you like). You could do that in multiple ways -- but the easiest way is to use _subject_goal_ instead of _subject_move_:

```rust
 score += match subject_goal {
   "X" => 0,
   "Y" => 3,
   "Z" => 6
   _ => unimplemented!()
 };
```